### PR TITLE
Add integration tests for P2P library - Closes #1068

### LINKED ISSUE #1068: Add more integration tests for P2P library
### Expected behavior

We need integration tests to cover the following features:

**Peer discovery**

- When the network starts all at once and disconnected without any seed nodes.
- When the network starts all at once and each node uses its next neighbour as a seed node.
- When network starts disconnected and where nodes are added one at a time; each node uses its previous neighbour as a seed node.
- When the network starts all at once; each node shares the same seed node.
- When the network starts disconnected and nodes are added one at a time; each node shares the same seed node.

**P2P.request**

- Should make request to the network; it should reach a single peer.
- Should be able to receive a response from the network; from a single peer.

**P2P.send**

- Should send a message to a subset of peers within the network; should reach multiple peers.

**P2P.applyNodeStatus**

- Should send our node status to a subset of peers within the network.

----

To make it easier to set up the tests (I.e. `beforeEach` and `afterEach` block), they should be divided based on network topology:

- Disconnected network
- Partially connected network
- Fully connected network

### Actual behavior

Missing integration tests.

### PR DESCRIPTION
**Pull request by jondubois**
### What was the problem?

Missing integration tests.

### How did I fix it?

- Implemented integration test cases.
- Fixed some bugs which were uncovered by the new test cases.

### How to test it?

`npm run test:integration`

### Review checklist

* The PR resolves #1068
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: packages/lisk-p2p/src/peer.ts [hunk 1]
```diff
@@ -201,8 +201,12 @@ export class Peer extends EventEmitter {
 		return this._wsPort;
 	}
 
-	public static constructPeerId(ipAddress: string, port: number): string {
-		return `${ipAddress}:${port}`;
+	public static constructPeerId(ipAddress: string, wsPort: number): string {
```
Comment (shuse2) [inline packages/lisk-p2p/src/peer.ts]: prefer to make those 2 function as pure function.

File: packages/lisk-p2p/test/integration/p2p.ts [hunk 2]
```diff
@@ -140,12 +149,11 @@ describe('Integration tests for P2P library', () => {
...
 			});
 
-			const peerStartPromises: ReadonlyArray<Promise<void>> = p2pNodeList.map(
-				p2p => p2p.start(),
-			);
-
-			await Promise.all(peerStartPromises);
-			await wait(500);
+			// Launch nodes one at a time with a delay between each launch.
+			for (const p2p of p2pNodeList) {
```
Comment (shuse2) [inline packages/lisk-p2p/test/integration/p2p.ts:153]: we don't need `tslint:disable-next-line` here?

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: packages/lisk-p2p/test/integration/p2p.ts (hunk 2 above)
Comment (jondubois) [reply packages/lisk-p2p/test/integration/p2p.ts:153]: For some reason `npm run lint` doesn't seem to mind about this in the integration tests.

**Comment by jondubois** [comment]
Rebased to fix conflicts.

**Review by shuse2 — APPROVED** [review]

**Pull request merged**